feature/1675 - Add MCO

MichaelDMcCracken opened the pull request:
Resolves #1675 

The purpose of this pull request is to add MCO.

Fechulo commented:
Hey @MichaelDMcCracken 

You requested a review for this around 7 months ago, sorry for the delay. It seems this is still very much a work in progress, since it's still missing a lot. Have a look at the [airport file standards document](https://github.com/openscope/openscope/blob/develop/documentation/airport-file-standards.md) for more info. Let me know if you have any questions / need any help.

MichaelDMcCracken commented:
Ready for review, but I need someone to update it to the correct version. Thanks.

Fechulo commented on assets/airports/kmco.json:
```diff
@@ -0,0 +1,6135 @@
+        "ALYNA1": {
+            "icao": "ALYNA1",
+            "name": "Alayna One",
+            "entryPoints": {
```
Missing BEAUX and SURFR transitions

Fechulo commented on assets/airports/kmco.json:
```diff
@@ -0,0 +1,6135 @@
+            "entryPoints": {
+                "ADLYN": ["ADLYN", "ELITE", ["MAYYR", "A240+"]],
+                "CATEE": ["CATEE", "ELITE", ["MAYYR", "A240+"]],
+                "COAXE": ["COAXE"]
```
COAXE should be moved into the body since all transitions pass through it

Fechulo commented on assets/airports/kmco.json:
```diff
@@ -0,0 +1,6135 @@
+        "JAFAR2": {
+            "icao": "JAFAR2",
+            "name": "Jafar Two",
+            "entryPoints": {
```
Missing BRKWL transition

MichaelDMcCracken replied:
This seems contradictory to the statement above about the COAXE fix. All transitions pass through BRKWL. So do they need to go in the body or entryPoints?

Fechulo commented on assets/airports/kmco.json:
```diff
@@ -0,0 +1,6135 @@
+        "JOKRS2": {
+            "icao": "JOKRS2",
+            "name": "Jokers Two",
+            "entryPoints": {
```
Missing FAZES and MOANS transitions

Fechulo commented on assets/airports/kmco.json:
```diff
@@ -0,0 +1,6135 @@
+        "LEESE3": {
+            "icao": "LEESE3",
+            "name": "Leese Three",
+            "entryPoints": {
```
Missing OCF transition

Fechulo commented on assets/airports/kmco.json:
```diff
@@ -0,0 +1,6135 @@
+            },
+            "body": ["OCF", "ALADN"],
+            "rwy": {
+              "KMCO17L": [["SHIMM", "A110|S250"], "LEESE"],
```
There are no restrictions at SHIMM as far as I can see so everything in the rwy section can be moved to the body.
Also after LEESE, missing ORL

Fechulo commented on assets/airports/kmco.json:
```diff
@@ -0,0 +1,6135 @@
+        "MINEE5": {
+            "icao": "MINEE5",
+            "name": "Minnie Five",
+            "entryPoints": {
```
Missing PIE transition

Fechulo commented on assets/airports/kmco.json:
```diff
@@ -0,0 +1,6135 @@
+            "icao": "MINEE5",
+            "name": "Minnie Five",
+            "entryPoints": {
+              "BOXKR": ["BOXKR", "GUMMY", "LAL", ["ANDRO", "A130"]],
```
Missing SYKES after BOXKR

Fechulo commented on assets/airports/kmco.json:
```diff
@@ -0,0 +1,6135 @@
+        "PRICY1": {
+            "icao": "PRICY1",
+            "name": "Pricy One",
+            "entryPoints": {
```
Missing BULKS and CHIVE transitions

Fechulo commented on assets/airports/kmco.json:
```diff
@@ -0,0 +1,6135 @@
+            "name": "Pricy One",
+            "entryPoints": {
+              "BAYNE": ["BAYNE"],
+              "CHRGE": ["CHRGE", "DAAWG", ["CHIVE", "A240+|A260-"], ["DECKS", "A210-"]],
```
DECKS should be at **or above** FL210

Fechulo commented on assets/airports/kmco.json:
```diff
@@ -0,0 +1,6135 @@
+            },
+            "body": [["BULKS", "A190+|A200-|S280"], ["PRICY", "A130+"], "BOCCE", "JUMBA"],
+            "rwy": {
+              "KMCO17L": [["TAYYE", "A100+|A120-|S250"], "ARHIP", ["GIRAF", "A80+"], "DVOLA", ["^LYENS", "A50|S220"], "#006"],
```
LYENS should be at 210kts

Fechulo commented on assets/airports/airportLoadList.js:
```diff
@@ -276,6 +276,12 @@ window.AIRPORT_LOAD_LIST = (function() { // eslint-disable-line wrap-iife
         },
+        {
+            icao: 'kmco',
+            level: 'beginner',
```
With a current rate of 74.8 acph, this should be changed to 'hard'

Fechulo commented on documentation/airport-guides/kmco.md:
```diff
@@ -0,0 +1,15 @@
+MCO is served primarily by RNAV departure and arrival procedures. RNAV STARs are highly automated and require little ATC intervention. In North Flow (35/36s) RNAV SIDs are RNAV-Off-the-Ground to the en route transition. In South Flow (17/18s), RNAV SIDs begin with a coded heading upon departure wihch requires radar vectors to the first fix.
+
+## Additional Resources
+<a href="https://zjxartcc.org/media/doc/ZJX-1.D_Jacksonville_ARTCC_MCO_ATCT_SOP_pC6lZjs.pdf" target="_blank">Vatsim ZJX — Orlando ATCT Procedures</a> 
```
This document might be more useful:
https://zjxartcc.org/media/doc/ZJX-10.G_Jacksonville_ARTCC_F11_SOP_FDlIm85.pdf

Fechulo commented on documentation/airport-guides/kmco.md:
```diff
@@ -0,0 +1,15 @@
+The airport serves as a hub for Silver Airways, an operating base for JetBlue, Southwest Airlines and Spirit Airlines, as well as a focus city for Frontier Airlines. Southwest is the airport's largest carrier by passengers carried. The airport is also a major international gateway for the mid-Florida region, with over 850 daily flights on 44 airlines. The airport also serves 135 domestic and inte ...
+
+## Airspace
+The MCO airspace is fairly large and covers multiple smaller airports from the surface to 16,000'.
```
Officially the F11 TRACON

Fechulo commented:
@MichaelDMcCracken There are also some restricted areas to the North, South and East of MCO. They might be outside the TRACON but are still significant and should be included

Fechulo commented:
@MichaelDMcCracken See also line 442 of kmco.json:
FSHUN is misspelled in the body and causes sim to crash

MichaelDMcCracken commented:
![image](https://user-images.githubusercontent.com/26231967/152618734-1f43ef32-16da-4482-a5aa-d5b0d554c762.png)

@Fechulo 
One issue I've noticed is that the aircraft on departure do not fly the exact assigned heading from the SID. It is only 1 degree off. They are set to fly 176° (from 17R) and 191° (from 18L), but fly 177° and 192° respectively. Even though this is 15° apart, the sim is still giving the red rings. What's the deal?

Fechulo commented:
@erikquinn might be able to answer that question better

MichaelDMcCracken commented:
> There are also some restricted areas to the North, South and East of MCO. They might be outside the TRACON but are still significant and should be included
@Fechulo 
I added the north and south restricted areas. I did not add the restricted areas to the east at Cape Canaveral because they are only active during rocket launch windows and the MZULO departure goes right through the middle. Also added the Disney World TFR. Everything should be ready to go!

erikquinn commented:
The heading discrepancy is likely working correctly. The heading in the chart is a ground track to fly, and not really a heading (ie, they should apply wine correction to follow a 166 track off 17L (from your example). Given the wind of 220@7, they will need to crab slightly to the right to counteract the right-to-left crosswind they experience. One degree for that sounds about right.

As for the rings when their headings are 15 degrees apart, I suspect that to be an issue of rounding-- where they're like 14.846294600479 degrees divergent. A few things to fix there, which could go in a new issue. We could (a) allow it if `> minDivergence -1` to account for rounding errors, and (b) allow 10 degrees divergence when on an RNAV SID. The latter might be tough, but the former is easy. Would solve your issue while also punishing someone if they happened to assign 010 and 024.

MichaelDMcCracken commented:
> The heading discrepancy is likely working correctly. The heading in the chart is a ground track to fly, and not really a heading (ie, they should apply wind correction to follow a 166 track off 17L (from your example). Given the wind of 220@7, they will need to crab slightly to the right to counteract the right-to-left crosswind they experience. One degree for that sounds about right.
> 
> As for the rings when their headings are 15 degrees apart, I suspect that to be an issue of rounding-- where they're like 14.846294600479 degrees divergent. A few things to fix there, which could go in a new issue. We could (a) allow it if `> minDivergence -1` to account for rounding errors, and (b) allow 10 degrees divergence when on an RNAV SID. The latter might be tough, but the former is easy. Would solve your issue while also punishing someone if they happened to assign 010 and 024.

Copy. Wasn't sure if the `sh` command gave you a _response_ of the ground track or actual heading. 

My guess is that it is also a rounding error, because if I tell the aircraft on the 177° heading to `t l 1` it instantly corrects the issue.
